### PR TITLE
(#1445) Cleanup around Scalar and Callable

### PR DESCRIPTION
**Pull request by victornoel**
This is for #1445.

On top of the requirements from the ticket, I took the liberty of simplifying the constructors of some of the classes that interacts with Scalar and Callable for consistency sake. It's mainly deleting stuffs in order to limit the responsibility of each class.

### CODE REVIEW

File: src/main/java/org/cactoos/func/RunnableOf.java [hunk 1]
```diff
@@ -23,79 +23,61 @@
...
-    }
-
-    /**
-     * Ctor.
-     * @param proc Encapsulated proc
-     */
-    public RunnableOf(final Proc<X> proc) {
-        this(proc, null);
-    }
+    private final Scalar<?> scalar;
```
Comment (andreoss) [inline src/main/java/org/cactoos/func/RunnableOf.java:44]: @victornoel  Why is it a `Scalar` and not a `Runnable`?

File: src/main/java/org/cactoos/proc/ProcOf.java [hunk 2]
```diff
@@ -70,7 +53,7 @@ public ProcOf(final Func<X, ?> fnc) {
      * Ctor.
      * @param prc The proc
      */
-    private ProcOf(final Proc<X> prc) {
+    public ProcOf(final Proc<X> prc) {
```
Comment (andreoss) [inline src/main/java/org/cactoos/proc/ProcOf.java:56]: @victornoel This type of ctor doesn't make much sense, why would one want to wrap pre-existing `Proc` with `ProcOf`?

File: src/test/java/org/cactoos/experimental/TimedTest.java [hunk 3]
```diff
@@ -84,7 +84,7 @@
         "PMD.AvoidThrowingRawExceptionTypes", "PMD.DoNotThrowExceptionInFinally"
     })
     void containsResults() throws Exception {
-        new RepeatedCallable<>(
+        new Repeated<>(
```
Comment (andreoss) [inline src/test/java/org/cactoos/experimental/TimedTest.java:87]: @victornoel Can this type of test be migrated to `@RepeatedTest` annotation?

**Review by andreoss — CHANGES_REQUESTED** [review]

File: src/main/java/org/cactoos/func/RunnableOf.java (hunk 1 above)
Comment (victornoel) [reply src/main/java/org/cactoos/func/RunnableOf.java:44]: @andreoss Runnable.run does not throw an exception so it makes it awkward to assign a scalar (from the constructor) into a  runnable (while the opposite is acceptable). Now that I think about it, I could have wrapped the scalar into unchecked at the constructor level. I will do that for consistency sake with other similar classes.

File: src/main/java/org/cactoos/proc/ProcOf.java (hunk 2 above)
Comment (victornoel) [reply src/main/java/org/cactoos/proc/ProcOf.java:56]: @andreoss it's mainly useful to force type a lambda: `new ProcOf(x -> { /* do something with x */ ; })`

File: src/test/java/org/cactoos/experimental/TimedTest.java (hunk 3 above)
Comment (victornoel) [reply src/test/java/org/cactoos/experimental/TimedTest.java:90]: @andreoss I've added a todo, thx for the idea

**Comment by victornoel** [comment]
@andreoss it should be alright now

**Comment by victornoel** [comment]
@rultor merge

**Comment by victornoel** [comment]
@rultor stop

**Comment by victornoel** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#1522](https://github.com/yegor256/cactoos/issues/1522) is not in the agenda of @andreoss[/z](https://www.0crat.com/u/andreoss), can't set title

**Comment by 0crat** [comment]
Job [#1522](https://github.com/yegor256/cactoos/issues/1522) is not in the agenda of @andreoss[/z](https://www.0crat.com/u/andreoss), can't set title

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @andreoss[/z](https://www.0crat.com/u/andreoss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
Code review was too long (10 days), architects (@victornoel) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by sereshqua** [comment]
@andreoss please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**